xrpl_api/src/types/transaction/common.rs

```diff
@@ -1,49 +1,211 @@
+    #[serde(rename = "date", skip_serializing_if = "Option::is_none")]
     pub date: Option<LedgerTimestamp>,
 
     /// Transaction hash
-    #[serde(rename = "hash")]
+    #[serde(rename = "hash", skip_serializing_if = "Option::is_none")]
     pub hash: Option<String>,
 
     /// The ledger index of the ledger that includes this transaction.
-    #[serde(rename = "ledger_index")]
+    #[serde(rename = "ledger_index", skip_serializing_if = "Option::is_none")]
     pub ledger_index: Option<u32>,
     /// If true, this data comes from a validated ledger version; if omitted or
     /// set to false, this data is not final.
-    #[serde(rename = "validated")]
+    #[serde(rename = "validated", skip_serializing_if = "Option::is_none")]
     pub validated: Option<bool>,
 
     /// Meta is present in transactions returned by https://xrpl.org/ledger.html and
     /// also <https://xrpl.org/tx.html>. In other API
     /// methods it is found outside (next to) the transaction field.
-    #[serde(rename = "meta", alias = "metaData")]
+    #[serde(
+        rename = "meta",
+        alias = "metaData",
+        skip_serializing_if = "Option::is_none"
+    )]
     pub meta: Option<Meta>,
 }
 
-// #[derive(Debug, Deserialize)]
-// pub struct Memo {
-//     #[serde(rename = "MemoData")]
-//     pub memo_data: Option<String>,
-// }
+#[derive(Default, Debug, Clone, PartialEq)]
+pub struct Memo {
+    pub memo_data: Option<String>,
+    pub memo_format: Option<String>,
+    pub memo_type: Option<String>,
+}
+
+impl<'de> Deserialize<'de> for Memo {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MemoObjRaw {
+            memo_data: Option<String>,
+            memo_format: Option<String>,
+            memo_type: Option<String>,
+        }
+
+        #[derive(Deserialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MemoElmRaw {
+            memo: MemoObjRaw,
+        }
+
+        let elm = MemoElmRaw::deserialize(deserializer)?;
+
+        Ok(Memo {
+            memo_data: elm.memo.memo_data,
+            memo_format: elm.memo.memo_format,
+            memo_type: elm.memo.memo_type,
+        })
+    }
+}
+
+impl Serialize for Memo {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MemoObjRaw<'a> {
+            #[serde(skip_serializing_if = "Option::is_none")]
+            memo_data: Option<&'a str>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            memo_format: Option<&'a str>,
+            #[serde(skip_serializing_if = "Option::is_none")]
+            memo_type: Option<&'a str>,
+        }
+
+        #[derive(Serialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct MemoElmRaw<'a> {
```

Comment:
What does 'Elm' stand for?

Reply:
Just `Element`, should have typed it out fully. It it they outer layer in the weird array representation with the Element(Object(..)) structure